experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -463,3 +500,130 @@ function Checkpointer.get_model_prog_state(sim::ClimaLandSimulation)
+models to get the total fluxes.
+
+Because the integrated land model is composed of multiple sub-components, the
+fluxes are computed for each sub-component and then combined to get the total.
+
+# Arguments
+- `csf`: [CC.Fields.Field] containing a NamedTuple of turbulent flux fields: `F_turb_ρτxz`, `F_turb_ρτyz`, `F_turb_energy`, `F_turb_moisture`.
+- `sim`: [ClimaLandSimulation] the integrated land simulation to compute fluxes for.
+- `atmos_sim`: [Interfacer.AtmosModelSimulation] the atmosphere simulation to compute fluxes with.
+- unused arguments: `boundary_space`, `thermo_params`, `surface_scheme`
+"""
+function FluxCalculator.compute_surface_fluxes!(
+    csf,
+    sim::ClimaLandSimulation,
+    atmos_sim::Interfacer.AtmosModelSimulation,
+    _...,
+)
+    coupled_atmos = sim.model.soil.boundary_conditions.top.atmos
+
+    # `_int` refers to atmos state of center level 1
+    z_int = Interfacer.get_field(atmos_sim, Val(:height_int))
+    uₕ_int = Interfacer.get_field(atmos_sim, Val(:uv_int))
+    thermo_state_int = Interfacer.get_field(atmos_sim, Val(:thermo_state_int))
+
+    # We use parent here because of the mismatch between the lowest atmos level
+    #  and the boundary space, even though the horizontal grids are the same.
+    @assert axes(coupled_atmos.u).grid == axes(uₕ_int).grid.full_grid.horizontal_grid
+    @assert axes(coupled_atmos.thermal_state).grid == axes(thermo_state_int).grid.full_grid.horizontal_grid
+    @assert axes(coupled_atmos.h).grid == axes(z_int).grid.full_grid.horizontal_grid
+    parent(coupled_atmos.u) .= parent(uₕ_int)
+    parent(coupled_atmos.thermal_state) .= parent(thermo_state_int)
+    parent(coupled_atmos.h) .= parent(z_int)
+
+    # set the same atmosphere state for all sub-components
+    @assert sim.model.soil.boundary_conditions.top.atmos ===
+            sim.model.canopy.boundary_conditions.atmos ===
+            sim.model.snow.boundary_conditions.atmos ===
+            coupled_atmos
+
+    # get area fraction (min = 0, max = 1)
+    area_fraction = Interfacer.get_field(sim, Val(:area_fraction))
+    # get area mask [0, 1], where area_mask = 1 if area_fraction > 0
+    area_mask = Utilities.binary_mask.(area_fraction)
+
+    Y, p, t = sim.integrator.u, sim.integrator.p, sim.integrator.t
+
+    # update `csf` in-place using this model's flux calculation
+    fluxes = coupler_land_turbulent_fluxes!(p, sim.model, Y, t, coupled_atmos)
+    (; F_turb_ρτxz, F_turb_ρτyz, F_shf, F_lhf, F_turb_moisture) = fluxes
+
+    # add the flux contributing from this surface to the coupler field
+    # note that the fluxes are area-weighted, so if a surface model is
+    #  not present at this point, the fluxes are zero
+    @. csf.F_turb_ρτxz += F_turb_ρτxz * area_fraction * area_mask
+    @. csf.F_turb_ρτyz += F_turb_ρτyz * area_fraction * area_mask
+    @. csf.F_turb_energy += (F_shf .+ F_lhf) * area_fraction * area_mask
+    @. csf.F_turb_moisture += F_turb_moisture * area_fraction * area_mask
```

Comment:
This is setting the coupler surface fields, but later on, are those used to update the land cache? Or are we using this `coupler_land_turbulent_fluxes!(p, sim.model, Y, t, coupled_atmos)` to update the land cache with the turb fluxes to be used for the land timestep?

Reply:
The land cache is updated in `coupler_land_turbulent_fluxes!`, so the coupler never updates the land's fluxes. This is different from every other component model which isn't very clean, but it would be redundant to update them again with the same values